File: include/extras-kwidgetsaddons/libkstyleextensions.zig
const QtC = @import("qt6zig");
const qtc = @import("qt6c");
const QWidget = @import("libqt6").QWidget;
const qstyle_enums = @import("../libqstyle.zig").enums;

/// ### [Upstream resources](https://api.kde.org/kstyleextensions.html)
pub const KStyleExtensions = extern struct {
    /// ### [Upstream resources](https://api.kde.org/kstyleextensions.html)
    ///
    /// The pointer to the underlying Qt C++ object
    ///
    ptr: QtC.KStyleExtensions,

    pub const _is_KStyleExtensions = {};

    /// ### [Upstream resources](https://api.kde.org/kstyleextensions.html#customControlElement)
    ///
    /// ## Parameter(s):
    ///
    /// ` element: []const u8 `
    ///
    /// ` widget: QWidget `
    ///
    /// ## Returns:
    ///
    /// ` qstyle_enums.ControlElement `
    ///
    pub fn CustomControlElement(element: []const u8, widget: anytype) i32 {
        const element_str = qtc.libqt_string{
            .len = element.len,
            .data = element.ptr,
        };
        comptime _ = @TypeOf(widget)._is_QWidget;
        return qtc.KStyleExtensions_CustomControlElement(element_str, @ptrCast(widget.ptr));
    }

    /// ### [Upstream resources](https://api.kde.org/kstyleextensions.html#customStyleHint)
    ///
    /// ## Parameter(s):
    ///
    /// ` element: []const u8 `
    ///
    /// ` widget: QWidget `
    ///
    /// ## Returns:
    ///
    /// ` qstyle_enums.StyleHint `
    ///
    pub fn CustomStyleHint(element: []const u8, widget: anytype) i32 {
        const element_str = qtc.libqt_string{
            .len = element.len,
            .data = element.ptr,
        };
        comptime _ = @TypeOf(widget)._is_QWidget;
        return qtc.KStyleExtensions_CustomStyleHint(element_str, @ptrCast(widget.ptr));
    }

    /// ### [Upstream resources](https://api.kde.org/kstyleextensions.html#customSubElement)
    ///
    /// ## Parameter(s):
    ///
    /// ` element: []const u8 `
    ///
    /// ` widget: QWidget `
    ///
    /// ## Returns:
    ///
    /// ` qstyle_enums.SubElement `
    ///
    pub fn CustomSubElement(element: []const u8, widget: anytype) i32 {
        const element_str = qtc.libqt_string{
            .len = element.len,
            .data = element.ptr,
        };
        comptime _ = @TypeOf(widget)._is_QWidget;
        return qtc.KStyleExtensions_CustomSubElement(element_str, @ptrCast(widget.ptr));
    }
};

File: src/extras-kwidgetsaddons/libkstyleextensions.cpp
#include <KStyleExtensions>
#include <QString>
#include <QWidget>
#include <kstyleextensions.h>
#include "libkstyleextensions.h"
#include "libkstyleextensions.hxx"

int KStyleExtensions_CustomControlElement(const libqt_string element, const QWidget* widget) {
    QString element_QString = QString::fromUtf8(element.data, element.len);
    return static_cast<int>(KStyleExtensions::customControlElement(element_QString, widget));
}

int KStyleExtensions_CustomStyleHint(const libqt_string element, const QWidget* widget) {
    QString element_QString = QString::fromUtf8(element.data, element.len);
    return static_cast<int>(KStyleExtensions::customStyleHint(element_QString, widget));
}

int KStyleExtensions_CustomSubElement(const libqt_string element, const QWidget* widget) {
    QString element_QString = QString::fromUtf8(element.data, element.len);
    return static_cast<int>(KStyleExtensions::customSubElement(element_QString, widget));
}
